lib/index.js

```diff
@@ -114,18 +103,58 @@ function setupRateLimitEventExt(server, options) {
   };
 
   server.ext(event, (request, reply) => {
+    const options = parseRouteOptions(request);
+    if (!options.enabled) {
+      return reply.continue();
+    }
     // This handler is going to be called one time per registration of patova
-    getType(request, reply, (err, type) => {
-      extractKeyAndTakeToken(options.limitd, request, reply, type);
+    getType(options, request, reply, (err, type) => {
+      extractKeyAndTakeToken(options, request, reply, type);
     });
   });
+
+  server.ext('onPreResponse', (request, reply) => {
+    const requestLimit = request.plugins.patova && request.plugins.patova.limit;
+
+    if (requestLimit && requestLimit.conformant) {
+      const options = parseRouteOptions(request);
+      if (options.addHeaders) {
+        const headers = new RateLimitHeaders(
+            requestLimit.limit,
+            requestLimit.remaining,
+            requestLimit.reset);
+
+        Object.keys(headers).forEach(
+          key => setResponseHeader(request, key, headers[key]));
+      }
+    }
+    reply.continue();
+  });
+
+  function parseRouteOptions(request) {
```

Comment:
this logic looks good. it is not really parsing though, right? I would change the function name to `getRouteOptions` maybe?